diff --git a/src/vs/workbench/workbench.common.main.ts b/src/vs/workbench/workbench.common.main.ts
index e7c16a7..5ede7d5 100644
--- a/src/vs/workbench/workbench.common.main.ts
+++ b/src/vs/workbench/workbench.common.main.ts
@@ -323,6 +323,9 @@ import './contrib/markdown/browser/markdown.contribution.js';
 // Keybindings Contributions
 import './contrib/keybindings/browser/keybindings.contribution.js';
 
+// Codex
+import './contrib/codexConductor/browser/codexConductor.contribution.js';
+
 // Snippets
 import './contrib/snippets/browser/snippets.contribution.js';
 
